specs/experimental/gov-token.md

```diff
@@ -0,0 +1,117 @@
+| Token name   | `Optimism`                                   |
+| Token symbol | `OP`                                         |
+
+`GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
+`ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
+voting system. The contract integrates with the `Alligator` contract to enable advanced delegations.
+
+### Token Minting
+
+`GovernanceToken` MUST have a `mint(address,uint256)` function with external visibility that allows the `owner()` address
+to mint an arbitrary number of new tokens to a specific address. This function MUST only be called by the contract
+owner, the `MintManager`, as enforced by the `onlyOwner` modifier inherited from the `Ownable` contract. When tokens
+are minted, the voting power of the recipient address MUST be updated accordingly in the `Alligator` contract via the
+`afterTokenTransfer` hook. The total token supply is capped to `2^208^ - 1` to prevent overflow risks in the voting
+system. If the total supply exceeds this limit, `_mint(address,uint256)`, as inherited from `ERC20Votes`, MUST revert.
+
+### Token Burning
+
+The contract MUST allow token holders to burn their own tokens using the inherited `burn(uint256)` or
+`burnFrom(address,uint256)` functions inherited from `ERC20Burnable`. When tokens are burned, the total supply and the
+holder's voting power MUST be reduced accordingly in the `Alligator` contract via the `afterTokenTransfer` hook. Burn functions
+MUST NOT allow a user's balance to underflow and attempts to reduce balance below zero MUST revert.
+
+### Voting Power
+
+Each token corresponds to one unit of voting power. Token holders who wish to utilize their tokens for voting MUST delegate
+their voting power to an address (can be their own address). An active delegation MUST NOT prevent a user from transferring
+tokens. The `GovernanceToken` contract MUST offer public accessor functions for querying voting power, as outlined below.
+
+#### Public Query Functions
+
+- `checkpoints(address,uint32) returns (Checkpoint)`
+  - MUST retrieve the n-th Checkpoint for a given address.
+- `numCheckpoints(address) returns (uint32)`
+  - MUST retrieve the total number of Checkpoint objects for a given address.
+- `delegates(address) returns (address)`
+  - MUST retrieve the address that an account is delegating to.
+- `getVotes() returns (uint256)`
+  - MUST retrieve the current voting power of an address.
+- `getPastVotes(address,uint256) returns (uint256)`
+  - MUST retrieve the voting power of an address at specific block number in the past.
+- `getPastTotalSupply(uint256) returns (uint256)`
+  - MUST return the total token supply at a specific block number in the past.
+
+Above query functions MUST apply the constraints defiend in [Delegation Query Forwarding](#delegation-query-forwarding).
+
+### Delegation
```

Comment:
Its not super clear here if all of the state modifying functions are included here since the spec doesn't go through them and define them 1 by 1 like other specs